base/db/R/insert.format.vars.R

```diff
@@ -122,32 +128,39 @@ insert.format.vars <- function(con, format_name, mimetype_id, notes = NULL, head
     formats_variables[is.na(formats_variables)] <- ""
 
     ###  udunit tests ###
-    for(i in 1:nrow(formats_variables)){
-      u1 <- formats_variables[1,"unit"]
-      u2 <- dplyr::tbl(con, "variables") %>% dplyr::select("id", units) %>% dplyr::filter(.data$id %in% !!formats_variables[[1, "variable_id"]]) %>% dplyr::pull("units")
-
-      if(!PEcAn.utils::unit_is_parseable(u1)){
-        PEcAn.logger::logger.error(
-          "Units '", u1,  "' not parseable.",
-          "Please provide a unit that is parseable by the udunits library."
-        )
-      }
-      # Grab the bety units and
-      if(!units::ud_are_convertible(u1, u2)){
-        PEcAn.logger::logger.error(
-          "Units are not convertable."
-        )
+    for(i in seq_len(nrow(formats_variables))){
+      u1 <- formats_variables[["unit"]][i]
+      vid <- formats_variables[["variable_id"]][i]
+      if(!is.na(u1) && u1 != ""){
+        u2 <- dplyr::tbl(con, "variables") %>%
+          dplyr::select(.data$id, .data$units) %>%
+          dplyr::filter(.data$id %in% !!vid) %>%
+          dplyr::pull(.data$units)
+
+        if(!PEcAn.utils::unit_is_parseable(u1)){
+          PEcAn.logger::logger.error(
+            "Units '", u1,  "' not parseable.",
+            "Please provide a unit that is parseable by the udunits library."
+          )
+        }
+        # Grab the bety units and compare only if we have both units
+        if(length(u2) > 0 && !is.na(u2)){
+          if(!units::ud_are_convertible(u1, u2)){
+            PEcAn.logger::logger.error(
+              "Units are not convertible."
+            )
+          }
+        }
       }
     }
   }
 
     formats_df <- tibble::tibble(
-      header = as.character(header),
+      header = header,
       skip = skip,
```

Comment:
Both `header` and `skip` need to be of type character when they get inserted into the `formats` table of BetyDB. As commented above, it _was_ a bit weird that `skip` was converted way up at the top of the function while `header` stayed logical until here. Let's coerce both in the same place:
```suggestion
      header = as.character(header),
      skip = as.character(skip),
```